### PR TITLE
Refactor KafkaWriter 

### PR DESCRIPTION
**Pull request by zzhhhzz**
Refactor KafkaWriter to attach headers that include hostname and locality information.

### CODE REVIEW

File: singer/src/main/java/com/pinterest/singer/writer/HeaderProvider.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import com.pinterest.singer.common.SingerSettings;
+import org.apache.kafka.common.header.Headers;
+
+public class HeaderProvider {
+
+  public static final byte[] HOSTNAME = SingerSettings.getEnvironment().getHostname().getBytes();
+  public static final byte[] LOCALITY = SingerSettings.getEnvironment().getLocality().getBytes();
+
+  public static void addEnvHeaders(Headers headers){
+    headers.add("hostname", HOSTNAME);
```
Comment (HenryCaiHaiying) [inline singer/src/main/java/com/pinterest/singer/writer/HeaderProvider.java]: Define those string as constants.

File: singer/src/main/java/com/pinterest/singer/writer/HeaderProvider.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.pinterest.singer.writer;
+
+import com.pinterest.singer.common.SingerSettings;
+import org.apache.kafka.common.header.Headers;
+
+public class HeaderProvider {
+
+  public static final byte[] HOSTNAME = SingerSettings.getEnvironment().getHostname().getBytes();
```
Comment (HenryCaiHaiying) [inline singer/src/main/java/com/pinterest/singer/writer/HeaderProvider.java]: Get we also get Cell Id from SingerSettings?

File: singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java [hunk 3]
```diff
@@ -221,19 +222,85 @@ private void includeAuditMessageInBatch(List<ProducerRecord<byte[], byte[]>> mes
...
     Set<Future<KafkaWritingTaskResult>> resultSet = new HashSet<>();
     KafkaProducer<byte[], byte[]> producer = KafkaProducerManager.getProducer(producerConfig);
     Preconditions.checkNotNull(producer);
-
     if (producerConfig.isTransactionEnabled()) {
       producer.beginTransaction();
     }
     try {
       List<PartitionInfo> partitions = producer.partitionsFor(topic);
+      /*
```
Comment (HenryCaiHaiying) [inline singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java]: Remove commented out code.

File: singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java [hunk 4]
```diff
@@ -221,19 +222,85 @@ private void includeAuditMessageInBatch(List<ProducerRecord<byte[], byte[]>> mes
     return buckets;
   }
 
+  public List<List<LogMessage>> logMessagesCollation(List<PartitionInfo> partitions,
+                                                     String topic,
+                                                     List<LogMessage> logMessages) throws Exception {
+    LOG.info("Collate " + logMessages.size() + " messages");
```
Comment (HenryCaiHaiying) [inline singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java]: Would this be too much logging?

File: singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java (hunk 3 above)
Comment (HenryCaiHaiying) [reply singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java]: Why do we need to break the previous methods into two methods?  Could header information be attached to the List<List<ProducerRecord>> generated from messageCollation() method?

File: singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java (hunk 4 above)
Comment (ambud) [reply singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java]: Default log level is can control the volume of logging

**Comment by zzhhhzz** [comment]
this PR is no longer needed.   newer implementation is https://github.com/pinterest/singer/pull/13

**Pull request closed without merging**